test/Test/Time/ReadShow.hs

```diff
@@ -12,55 +10,39 @@ module Test.Time.ReadShow
 
-import Time (DayUnit, FortnightUnit, HourUnit, KnownRat, MicroSecondUnit, MilliSecondUnit,
-             MinuteUnit, NanoSecondUnit, Rat, RatioNat, SecondUnit, ShowUnit, Time (..), WeekUnit)
+import Time (DayUnit, FortnightUnit, HourUnit, MicroSecondUnit, MilliSecondUnit, MinuteUnit,
+             NanoSecondUnit, RatioNat, SecondUnit, ShowUnit, Time (..), WeekUnit)
 
 import qualified Hedgehog.Gen as Gen
 import qualified Hedgehog.Range as Range
 
--- | GADT for time units.
-data Unit (multiplier :: Rat) where
-    USecond      :: Unit SecondUnit
-    UMilliSecond :: Unit MilliSecondUnit
-    UMicroSecond :: Unit MicroSecondUnit
-    UNanoSecond  :: Unit NanoSecondUnit
-    UMinute      :: Unit MinuteUnit
-    UHour        :: Unit HourUnit
-    UDay         :: Unit DayUnit
-    UWeek        :: Unit WeekUnit
-    UFortnight   :: Unit FortnightUnit
+readShowTestTree :: TestTree
+readShowTestTree = testProperty "Hedgehog read . show == id" prop_readShowUnit
 
 -- | Existential data type for 'Unit's.
-data AnyTime = forall unit r . (KnownRat r, KnownSymbol (ShowUnit r), KnownSymbol (ShowUnit unit)) => Time unit ::: Unit r
+data AnyTime =  forall unit . (KnownSymbol (ShowUnit unit))
+             => MkAnyTime (Time unit)
 
 -- | Returns the 'AnyTime' depending on given (random) integer.
 unitChooser :: Int -> RatioNat -> AnyTime
-unitChooser 0 t = Time @SecondUnit t ::: USecond
-unitChooser 1 t = Time @MilliSecondUnit t ::: UMilliSecond
-unitChooser 2 t = Time @MicroSecondUnit t ::: UMicroSecond
-unitChooser 3 t = Time @NanoSecondUnit t ::: UNanoSecond
-unitChooser 4 t = Time @MinuteUnit t ::: UMinute
-unitChooser 5 t = Time @HourUnit t ::: UHour
-unitChooser 6 t = Time @DayUnit t ::: UDay
-unitChooser 7 t = Time @WeekUnit t ::: UWeek
-unitChooser 8 t = Time @FortnightUnit t ::: UFortnight
+unitChooser 0 t = MkAnyTime $ Time @SecondUnit      t
+unitChooser 1 t = MkAnyTime $ Time @MilliSecondUnit t
+unitChooser 2 t = MkAnyTime $ Time @MicroSecondUnit t
+unitChooser 3 t = MkAnyTime $ Time @NanoSecondUnit  t
+unitChooser 4 t = MkAnyTime $ Time @MinuteUnit      t
+unitChooser 5 t = MkAnyTime $ Time @HourUnit        t
+unitChooser 6 t = MkAnyTime $ Time @DayUnit         t
+unitChooser 7 t = MkAnyTime $ Time @WeekUnit        t
+unitChooser 8 t = MkAnyTime $ Time @FortnightUnit   t
 unitChooser _ _ = error "Impossible happened"
 
 -- | Verifier for 'AnyTime'.
 verifyAnyTime :: (MonadTest m) => AnyTime -> m ()
-verifyAnyTime (t ::: USecond)      = verifyProperty t
-verifyAnyTime (t ::: UMilliSecond) = verifyProperty t
-verifyAnyTime (t ::: UMicroSecond) = verifyProperty t
-verifyAnyTime (t ::: UNanoSecond)  = verifyProperty t
-verifyAnyTime (t ::: UMinute)      = verifyProperty t
-verifyAnyTime (t ::: UHour)        = verifyProperty t
-verifyAnyTime (t ::: UDay)         = verifyProperty t
-verifyAnyTime (t ::: UWeek)        = verifyProperty t
-verifyAnyTime (t ::: UFortnight)   = verifyProperty t
+verifyAnyTime (MkAnyTime t) = verifyProperty t
```

Comment:
Probably you don't need separate `verifyProperty` function in this case since you have only single case. But that's minor.